server/server.py

```diff
@@ -0,0 +1,61 @@
+import os
+import pandas as pd
+from flask import Flask, request, send_file, jsonify
+from metadata import MetadataDict, Metadata
+
+HOST = '127.0.0.1'
+PORT = '5000'
+
+app = Flask(__name__)
+file_metadata = MetadataDict()
+
+@app.route('/upload', methods=['POST'])
+def handle_file_upload_request():
+
+    uploaded_file = request.files['file']
+    uploaded_file.save(uploaded_file.filename)
+
+    file_metadata.store(
+        uploaded_file.filename,
+        Metadata(uploaded_file.filename)
+    )
+
+
+@app.route('/filelist', methods=['GET'])
+def handle_file_list_request():
+
+    return jsonify(list(filter(
+        lambda filename: filename.endswith('.csv'),
+        os.listdir()
+    )))
+
+
+@app.route('/metadata', methods=['GET'])
+def handle_file_metadata_request():
+
+    filename = request.args.get('filename')
+    print(file_metadata.metadata_dict.keys())
```

Comment:
Remove the print statement as it accesses the dictionary object's keys without locking it.

OR

please use locks